src/helpers/modeRegistry.ts

```diff
@@ -31,6 +33,24 @@ class ModeRegistry {
         });
     }
 
+    static modesRegisterFromDir(dirPath: string): void {
+        const modes: any[] = [];
+
+        const files = fs.readdirSync(dirPath);
```

Comment:
Consider adding error handling for file operations in the `modesRegisterFromDir` function. This can help to catch and handle potential issues such as file access permissions or non-existent directories. [important]